pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -741,3 +743,22 @@ func CheckMissingResultReferences(pipelineRunState PipelineRunState, targets Pip
 	}
 	return nil
 }
+
+// createResultsCacheMatrixedTaskRuns creates a cache of results that have been fanned out from a
+// referenced matrixed PipelintTask so that you can easily access these results in subsequent Pipeline Tasks
+func createResultsCacheMatrixedTaskRuns(rpt *ResolvedPipelineTask) (resultsCache map[string][]string) {
+	if len(rpt.ResultsCache) == 0 {
+		resultsCache = make(map[string][]string)
+	}
+	// Sort the taskRuns by name to ensure the order is deterministic
+	sort.Slice(rpt.TaskRuns, func(i, j int) bool {
+		return rpt.TaskRuns[i].Name < rpt.TaskRuns[j].Name
+	})
```

Comment:
why do we need to sort this? Will it change the order of taskruns since we're passing a pointer?

Reply:
Yes, the order can change if we don't sort the slices.

Reply:
why the order matters here?

Reply:
The issue is that if you don't sort them here the child references are returned out of order and fail the unit tests: https://github.com/tektoncd/pipeline/blob/542c3b36307b9713fe0897aba726df12cbf537b0/pkg/reconciler/pipelinerun/pipelinerun_test.go#L12687

Reply:
So this is only for test purpose?

Reply:
If so I don't think it is a good idea that we have functional code to meet the test purpose. Can you sort it in tests instead?

Reply:
It uses cmpopts.SortSlices(lessChildReferences), which should sort the ChildRefs, but for some reason is not. 
https://github.com/tektoncd/pipeline/blob/542c3b36307b9713fe0897aba726df12cbf537b0/pkg/reconciler/pipelinerun/pipelinerun_test.go#L13074

Also tried sorting the TaskRuns within the test, but it seems like this logic is necessary in the code in order to get the correct Params mapped within the TaskSpec itself.